refactor: ♻️ remove setuptools check_requirements in executorch export

onuralpszr opened the pull request:
<!--
Thank you 🙏 for your contribution to [Ultralytics](https://www.ultralytics.com/) 🚀! Your effort in enhancing our repositories is greatly appreciated. To streamline the process and assist us in integrating your Pull Request (PR) effectively, please follow these steps:

1. Check for Existing Contributions: Before submitting, kindly explore existing PRs to ensure your contribution is unique and complementary.
2. Link Related Issues: If your PR addresses an open issue, please link it in your submission. This helps us better understand the context and impact of your contribution.
3. Elaborate Your Changes: Clearly articulate the purpose of your PR. Whether it's a bug fix or a new feature, a detailed description aids in a smoother integration process.
4. Ultralytics Contributor License Agreement (CLA): To uphold the quality and integrity of our project, we require all contributors to sign the CLA. Please confirm your agreement by commenting below:

    I have read the CLA Document and I sign the CLA

For more detailed guidance and best practices on contributing, refer to our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Your adherence to these guidelines ensures a faster and more effective review process.
-->

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Improves ExecuTorch export reliability on ARM64 Docker by conditionally adding a `packaging` dependency workaround 🐳🛠️

### 📊 Key Changes
- Added `IS_DOCKER` environment detection to the exporter utilities.
- Updated `export_executorch()` dependency checks:
  - Removed the global `setuptools<71.0.0` pin previously used as a workaround.
  - Added a targeted fix: on **Linux + ARM64 + Docker**, require `packaging>=22.0` to avoid ExecuTorch build issues.
- Kept ExecuTorch export requirements pinned to `executorch==1.0.1` and `flatbuffers`, and maintained `numpy<=2.3.5` for compatibility.

### 🎯 Purpose & Impact
- Improves **ExecuTorch export stability** specifically for users building/exporting in **ARM64 Docker environments** (common in CI and containerized deployments) 🧩✅
- Reduces unnecessary dependency constraints for other platforms by removing a broad `setuptools` pin, helping avoid version conflicts 🧹📦
- Makes the workaround more precise and future-proof, while preserving known compatibility pins needed for successful exports 🔒🚀

onuralpszr committed:
refactor: ♻️ remove setuptools check_requirements in executorch export

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

UltralyticsAssistant commented on ultralytics/engine/exporter.py:
```diff
@@ -1189,7 +1189,7 @@ def export_executorch(self, prefix=colorstr("ExecuTorch:")):
         # TorchAO release compatibility table bug https://github.com/pytorch/ao/issues/2919
         # Setuptools bug: https://github.com/pypa/setuptools/issues/4483
-        check_requirements("setuptools<71.0.0")  # Setuptools bug: https://github.com/pypa/setuptools/issues/4483
+        # check_requirements("setuptools<71.0.0")  # Setuptools bug: https://github.com/pypa/setuptools/issues/4483
```
💡 **MEDIUM**: Commenting out this guard may reintroduce a known failure mode during ExecuTorch export on environments with newer setuptools. If the pin is intentionally being removed, consider replacing it with a targeted runtime check that raises a clear error only when the problematic setuptools versions are detected (and ideally only for the code path that triggers the issue), so users get actionable guidance instead of a later, harder-to-diagnose failure.

UltralyticsAssistant reviewed (commented):
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, minimal refactor that removes a fragile dependency pin in the ExecuTorch export path while preserving the other important requirement checks. Only concern is loss of a known guardrail: without an alternative check or clearer failure messaging, users on affected setuptools versions may hit less actionable errors later in the export.

💬 Posted 1 inline comment

onuralpszr committed:
refactor: ♻️ remove obsolete setuptools compatibility checks.

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

onuralpszr commented:
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2025-12-27 16:51:23 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/20541723134): `ci.yml` run 15885
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/20541723317): `docker.yml` run 1557

UltralyticsAssistant reviewed (approved):
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

✅ Clean, minimal change. Commenting out the `setuptools<71.0.0` runtime pin in `export_executorch()` should not introduce functional regressions and reduces fragile dependency enforcement during export. The remaining checks (`torch>=2.9.0`, `executorch==1.0.1`, `flatbuffers`, and `numpy<=2.3.5`) still provide a clear dependency guardrail. No actionable issues found in this diff.

onuralpszr committed:
refactor: ♻️ remove arm64 uv check in requirements auto-update

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

onuralpszr committed:
refactor: ♻️ revert ARM64 check for uv usage in requirements auto-update

onuralpszr committed:
chore: 💥 debug logger for failing docker executorch check

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

onuralpszr commented:
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2025-12-28 01:05:01 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/20546733214): `ci.yml` run 15897
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/20546733430): `docker.yml` run 1559

onuralpszr committed:
chore: 💥 revert debug logger for failing docker executorch check

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

onuralpszr committed:
chore: add setuptools requirement check before ExecuTorch export and remove lower limit instalation of setuptools

Signed-off-by: Onuralp SEZER <onuralp@ultralytics.com>

onuralpszr commented:
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2025-12-28 02:21:01 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/20547489294): `ci.yml` run 15901
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/20547489491): `docker.yml` run 1561

lakshanthad reviewed (commented):
@onuralpszr I have tested locally on Raspberry Pi and tests are passing. However, I can see one arm64 Docker test is failing at the moment.

onuralpszr committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad committed:
Remove `setuptools` and add `packaging` pin

lakshanthad commented:
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2025-12-30 00:37:10 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/20585906331): `ci.yml` run 15966
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/20585906572): `docker.yml` run 1571

lakshanthad committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

onuralpszr committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad reviewed (commented):
@onuralpszr I have removed `setuptools` and added `packaging>=22.0` pin. Fast, slow CIs and Docker are all passing now. Feel free to change the title of the PR if this change is good :)

onuralpszr commented:
> @onuralpszr I have removed `setuptools` and added `packaging>=22.0` pin. Fast, slow CIs and Docker are all passing now. Feel free to change the title of the PR if this change is good :)

Maybe we can set to install packaging only for ARM because rest doesn't needed ? Once we fix docker side (I am still thinking it" so we can remove all together as well. This will minimize installation in others. 

Also thank you very much for debugging it.

onuralpszr committed:
Merge branch 'main' into feat/export-executorch-dep-setuptools

lakshanthad committed:
Scope `packaging` to Linux Arm64 Docker

onuralpszr committed:
Merge fa4978e59c68cd91a6d00061ed314061638aca56 into e3a987c2a74c4ab55e0fbef3b0d8f993cc91c198

UltralyticsAssistant committed:
Auto-update Ultralytics Docs Reference by https://ultralytics.com/actions

lakshanthad reviewed (approved):
@onuralpszr Everything is passing now. @glenn-jocher This PR is good to merge!
